modules/git/parse_nogogit_test.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+// +build !gogit
+
+package git
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseTreeEntries(t *testing.T) {
+	testCases := []struct {
+		Input    string
+		Expected []*TreeEntry
+	}{
+		{
+			Input:    "",
+			Expected: []*TreeEntry{},
+		},
+		{
+			Input: `100644 blob ea0d83c9081af9500ac9f804101b3fd0a5c293af	README.md
+100644 blob 037f27dc9d353ae4fd50f0474b2194c593914e35	README_ZH.md
+040000 tree 84b90550547016f73c5dd3f50dea662389e67b6d	assets
+100644 blob 9846a94f7e8350a916632929d0fda38c90dd2ca8	atters.md
+`,
+			Expected: []*TreeEntry{
+				{
+					ID:        MustIDFromString("ea0d83c9081af9500ac9f804101b3fd0a5c293af"),
```

Comment:
I'm not sure if we shold test against main repo ...

Reply:
it's not testing against the main repo - these are synthetic tests that just so happen to have been created from the main repo.

Reply:
Wont it change if Readme changes too?

Reply:
it doesn't look at files - it's just testing the string.